### PR TITLE
add functions to get api call counts and total cpu usage

### PR DESCRIPTION
**Pull request by surabhil**
- Gets the api call count from Usage Log
- Gets the total cpu payment for an api

### CODE REVIEW

File: src/instrument.js [hunk 1]
```diff
@@ -1,6 +1,7 @@
 const APIM_CONTRACT_NAME = 'manager.apim'
 const INSTR_CONTRACT_NAME = 'instr.ore'
-const INSTR_TABLE_NAME = 'tokens'
+const INSTR_USAGE_CONTRACT_NAME = 'usagelog.ore'
+const TABLE_NAME = 'instruments'
```
Comment (blackwatertepes) [inline src/instrument.js]: Are we using this anywhere?

File: src/instrument.js [hunk 2]
```diff
@@ -10,12 +11,22 @@ async function getAllInstruments(oreAccountName, additionalFilters = []) {
 
   const rows = await this.getAllTableRowsFiltered({
     code: INSTR_CONTRACT_NAME,
-    table: INSTR_TABLE_NAME,
+    table: 'tokens',
```
Comment (blackwatertepes) [inline src/instrument.js]: We should move this back to a CONSTANT.

File: src/instrument.js [hunk 3]
```diff
@@ -51,44 +65,62 @@ async function findInstruments(oreAccountName, activeOnly = true, category = und
...
   // Confirms that issuer in Instrument matches signature of transaction
   // Creates an instrument token, populate with params, save to issuer account
   // Saves endpoints to endpoints_published
   let options = {authorization: `${oreAccountName}@active`}
   let contract = await this.eos.contract(APIM_CONTRACT_NAME, options)
-  instrument.start_time = instrument.start_time || Date.now()
-  instrument.end_time = instrument.end_time || Date.now() + ONE_YEAR
-  await contract.publishapi(oreAccountName, instrument.apiName, instrument.rights, instrument.description, instrument.start_time, instrument.end_time, options)
+  let start_time = Date.now()
+  let end_time = Date.now() + ONE_YEAR
```
Comment (blackwatertepes) [inline src/instrument.js]: Any reason why we don't want to allow for custom start_time & end_time?

File: src/instrument.js [hunk 4]
```diff
@@ -51,44 +65,62 @@ async function findInstruments(oreAccountName, activeOnly = true, category = und
...
+}
+
 module.exports = {
   exerciseInstrument,
   findInstruments,
   getInstruments,
-  getRight,
-  saveInstrument
+  saveInstrument,
+  getAPiCallCount,
```
Comment (blackwatertepes) [inline src/instrument.js]: `getApiCallCount`

File: src/instrument.js (hunk 1 above)
Comment (surabhil) [reply src/instrument.js]: not using "instruments" table 
using "tokens" table to get all the instruments

File: src/instrument.js (hunk 2 above)
Comment (surabhil) [reply src/instrument.js]: done

File: src/instrument.js (hunk 3 above)
Comment (surabhil) [reply src/instrument.js]: we do..don't need it here. We can pass these when we create instruments.

File: src/instrument.js (hunk 4 above)
Comment (surabhil) [reply src/instrument.js]: done

File: src/instrument.js [hunk 5]
```diff
@@ -51,44 +67,46 @@ async function findInstruments(oreAccountName, activeOnly = true, category = und
...
+  let options = {authorization: `${oreAccountName}@active`}
+  let contract = await this.eos.contract(APIM_CONTRACT_NAME, options)
+  let voucher = await contract.licenceapi(oreAccountName, offerInstrumentId, options)
+
+  return voucher
+}
+
+async function getApiCallStats(rightName){
+  //calls the usagelog contract to get the total number of calls against a particular right
+  let calls = await eos.getAllTableRows({
```
Comment (blackwatertepes) [inline src/instrument.js]: I think this should be `this.getAllTableRows`

File: src/instrument.js [hunk 6]
```diff
@@ -51,44 +67,46 @@ async function findInstruments(oreAccountName, activeOnly = true, category = und
...
-}
-
 async function saveInstrument(oreAccountName, instrument) {
   // Confirms that issuer in Instrument matches signature of transaction
   // Creates an instrument token, populate with params, save to issuer account
   // Saves endpoints to endpoints_published
   let options = {authorization: `${oreAccountName}@active`}
   let contract = await this.eos.contract(APIM_CONTRACT_NAME, options)
-  instrument.start_time = instrument.start_time || Date.now()
-  instrument.end_time = instrument.end_time || Date.now() + ONE_YEAR
```
[the remark this answers is not in the document]
Comment (surabhil) [reply src/instrument.js]: we have default values defined in the contract.

[the remark this answers is not in the document]
Comment (blackwatertepes) [reply src/instrument.js]: 👍

**Pull request merged**